docs/core/deploying/native-aot/diagnostics.md

```diff
@@ -0,0 +1,54 @@
+
+- Trimming as a first-class feature in NativeAOT application. Unused code in a NativeAOT application is stripped out of the final binary. Any unbounded reflection used by the application including its dependent libraries will be trimmed. NativeAOT compiler will generate warnings in such cases, and it is critical that these warnings are fixed. For example, suppressing any such warnings without careful analysis can lead to hard to debug production failures.
+- Symbol file issues: In NativeAOT, symbol-file-dependent diagnostics (debugging, PerfView callstacks, etc) don't work at all unless the diagnostic tool has access to the monolithic PDB that was generated when the application was compiled. In .NET, symbol-file-dependent diagnostics generally work just fine (or even work great) even if the diagnostic tool has no access to any PDBs that were generated when the application was compiled (i.e., symbols for the .NET runtime and the .NET libraries can be pulled from symbol servers, diagnostic tools can still show function names, accurate callstacks, etc even without access to the PDBs generated at application compile time).
+- NativeAOT application is neither a typical managed application (for example, no JIT) nor a typical native application (for example, has full GC support). The intent is to meet the reasonable diagnostics expectation of a product that is coming out of the .NET family and also the experience should feel familiar to someone who does production diagnostics on a C++ application.
+
+## Current NativeAOT diagnostic support
+
+Not all diagnostic features in NativeAOT are currently at parity with .NET. The following table summarizes NativeAOT diagnostic feature parity with .NET:
+
+| Feature | Fully Supported | Partially Supported | Not Supported |
+| - | - | - | - |
+| Build (Inner dev loop) diagnostics | <span aria-hidden="true">✔️</span><span class="visually-hidden">Fully supported</span> | | |
+| Observability | | <span aria-hidden="true">✔️</span><span class="visually-hidden">Partially supported</span> | |
+| Profiling | | <span aria-hidden="true">✔️</span><span class="visually-hidden">Partially supported</span> | |
+| Production debugging | | <span aria-hidden="true">✔️</span><span class="visually-hidden">Partially supported</span> | |
+| Heap analysis | | | <span aria-hidden="true">❌</span><span class="visually-hidden">Not supported</span> |
+
+### Build time diagnostics
+
+Build time (inner-dev-loop experience) diagnostic experience for a NativeAOT application will match exactly the .NET application experience since the application in build-time will run on .NET. Given the challenges in acquiring diagnostics in production environment and the current gap in diagnostic tools in NativeAOT, spending time in build time diagnostic in NativeAOT will be the best solution to build a performant and reliable NativeAOT application.
+
+### Observability support
```

Comment:
I'd prefer the headings for different categories of support to exactly match the wording in the table